schema-registry/common/src/test/java/com/hortonworks/registries/schemaregistry/avro/AvroNestedCheckerTest.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2016 Hortonworks.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hortonworks.registries.schemaregistry.avro;
+
+import org.apache.avro.Schema;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.IOException;
+
+public class AvroNestedCheckerTest {
```

Comment:
Could you add all the tests in this call to "AvroCompositeSchemasTest"? AvroCompositeSchemasTest class is used for testing complex schemas, the tests in this class would be a good fit there.

Reply:
My only concern here, is that the composite aspect of schemas isn't Avro specific AFAIK; it's something Hortonworks added. Where the ability to nest records inside of them selves is standard Avro.

It seems to me that they're separate problems (even if they sometimes leverage the same code path), and should get tested separately.

Let me know what you think; I can change it so they're grouped together if that's preferred.

Reply:
@Wesley-Lawrence: Yeah that makes sense. I agree we should keep them separate.